crates/nargo_cli/Cargo.toml

```
@@ -50,5 +53,15 @@ predicates = "2.1.5"
 [features]
 default = ["plonk_bn254"]
 # The plonk backend can only use bn254, so we do not specify the field
-plonk_bn254 = ["acvm-backend-barretenberg/native"]
-plonk_bn254_wasm = ["acvm-backend-barretenberg/wasm"]
+plonk_bn254 = [
+    "acvm-backend-barretenberg/native",
+    "async-lsp/client-monitor",
+    "async-lsp/stdio",
+    "async-lsp/tracing"
+]
+plonk_bn254_wasm = [
+    "acvm-backend-barretenberg/wasm",
+    "async-lsp/client-monitor",
+    "async-lsp/stdio",
+    "async-lsp/tracing"
+]
```

```suggestion
plonk_bn254 = ["acvm-backend-barretenberg/native", "lsp"]
plonk_bn254_wasm = ["acvm-backend-barretenberg/wasm", "lsp"]
lsp = [
    "async-lsp/client-monitor",
    "async-lsp/stdio",
    "async-lsp/tracing"
]
```

We should factor out these flags to keep them consistent.

---

I think this is bad because it advertises that `lsp` is a feature that can be enabled. If you don't like the duplication, we have to remove the `.workspace = true` from the dep and instead specify it with the features itself.

---

I split the deps.

---

Yeah, this is a bit unfortunate. Ideally Cargo would allow us to have internal feature flags.

Happy to split deps as this isn't going to spread out over many crates.